apps/system/js/bluetooth.js

```diff
@@ -48,6 +48,10 @@ var Bluetooth = {
 
   /* this property store a reference of the default adapter */
   defaultAdapter: null,
+  /* to host single adapter promise for Bluetooth:adapter */
+  _getAdapterPromise: null,
+  /* to resolve adapter request for Bluetooth:adapter */
+  _resolveAdapterPromise: null,
```

Comment:
I go through the promise code twice. In fact, it's a defer to resolve the promise. And I do a test to make sure that a promise is possible to run in `then()` function twice. The answer is not. So, we can only handle `_getAdapterPromise` to be cleaned in right time as you have done:)

But one thing is confusing me. The naming `_resolveAdapterPromise`. Sometimes, I misunderstand to identify `_resolveAdapterPromise`. It let me think it is a promise. So I would like to suggest you to rename it with `_getAdapterPromiseResolve`.